### PR TITLE
librealsense: 2.45.0 -> 2.54.2

### PR DESCRIPTION
**Pull request by bouk**
## Description of changes

Upgrade librealsense to 2.54.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/librealsense/default.nix [hunk 1]
```diff
@@ -42,22 +42,9 @@ stdenv.mkDerivation rec {
...
   patches = [
-    # fix build on aarch64-darwin
-    # https://github.com/IntelRealSense/librealsense/pull/9253
-    (fetchpatch {
-      url = "https://github.com/IntelRealSense/librealsense/commit/beb4c44debc8336de991c983274cad841eb5c323.patch";
-      sha256 = "05mxsd2pz3xrvywdqyxkwdvxx8hjfxzcgl51897avz4v2j89pyq8";
-    })
-    ./py_sitepackage_dir.patch
     ./py_pybind11_no_external_download.patch
   ];
```
Comment (pbsds) [inline pkgs/development/libraries/librealsense/default.nix:47]:
```suggestion

  preInstall = ''
    export HOME="$(mktemp -d)"
  '';

```

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 273539` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>librealsense-gui</li>
    <li>librealsense-gui.dev</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>librealsense (librealsenseWithoutCuda)</li>
    <li>librealsense.dev (librealsenseWithoutCuda.dev)</li>
    <li>librealsenseWithCuda</li>
    <li>librealsenseWithCuda.dev</li>
    <li>python310Packages.pyrealsense2 (python310Packages.pyrealsense2WithoutCuda)</li>
    <li>python310Packages.pyrealsense2.dev (python310Packages.pyrealsense2WithoutCuda.dev)</li>
    <li>python310Packages.pyrealsense2WithCuda</li>
    <li>python310Packages.pyrealsense2WithCuda.dev</li>
    <li>python311Packages.pyrealsense2 (python311Packages.pyrealsense2WithoutCuda)</li>
    <li>python311Packages.pyrealsense2.dev (python311Packages.pyrealsense2WithoutCuda.dev)</li>
    <li>python311Packages.pyrealsense2WithCuda</li>
    <li>python311Packages.pyrealsense2WithCuda.dev</li>
  </ul>
</details>

```
librealsense-gui> CMake Error at tools/realsense-viewer/cmake_install.cmake:82 (file):
librealsense-gui>   file INSTALL cannot make directory
librealsense-gui>   "/homeless-shelter/Documents/librealsense2/presets": No such file or
librealsense-gui>   directory.
librealsense-gui> Call Stack (most recent call first):
librealsense-gui>   tools/cmake_install.cmake:55 (include)
librealsense-gui>   cmake_install.cmake:192 (include)
```

File: pkgs/development/libraries/librealsense/default.nix (hunk 1 above)
Comment (pbsds) [reply pkgs/development/libraries/librealsense/default.nix:47]: or if possible, put the presets in `$out/share/librealsense`

Comment (bouk) [reply pkgs/development/libraries/librealsense/default.nix:48]: Thanks for the review, fixed

**Comment by pbsds** [comment]
@ofborg eval

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 273539` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>librealsense (librealsenseWithoutCuda)</li>
    <li>librealsense-gui</li>
    <li>librealsense-gui.dev</li>
    <li>librealsense.dev (librealsenseWithoutCuda.dev)</li>
    <li>librealsenseWithCuda</li>
    <li>librealsenseWithCuda.dev</li>
    <li>python310Packages.pyrealsense2 (python310Packages.pyrealsense2WithoutCuda)</li>
    <li>python310Packages.pyrealsense2.dev (python310Packages.pyrealsense2WithoutCuda.dev)</li>
    <li>python310Packages.pyrealsense2WithCuda</li>
    <li>python310Packages.pyrealsense2WithCuda.dev</li>
    <li>python311Packages.pyrealsense2 (python311Packages.pyrealsense2WithoutCuda)</li>
    <li>python311Packages.pyrealsense2.dev (python311Packages.pyrealsense2WithoutCuda.dev)</li>
    <li>python311Packages.pyrealsense2WithCuda</li>
    <li>python311Packages.pyrealsense2WithCuda.dev</li>
  </ul>
</details>


image

LGTM!

**Pull request merged**